### PR TITLE
Feature#83_Create_tests_for_all_models

### LINKED ISSUE #92: Tests for Order model

### LINKED ISSUE #144: Tests for shipping model

### PR DESCRIPTION
**Pull request by CheshirLvova**
Feature#144-tests_for_shipping_model

### CODE REVIEW

File: ProjectShop/order/tests/tests.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from django.test import TestCase
+from django.db.utils import IntegrityError
+
```
Comment (strangerok1507) [inline ProjectShop/order/tests/tests.py:3]: rename this file to test_models

File: ProjectShop/Shipping/tests/tests.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from django.test import TestCase
+from django.db.utils import IntegrityError
+
```
Comment (strangerok1507) [inline ProjectShop/Shipping/tests/tests.py:3]: rename this file to test_models

File: ProjectShop/order/tests/factories.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    payment_status = factory.Faker('random_int', min=0, max=1)
+
+class OrderItemsModelFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = OrderItemsModel
+
+    order = factory.SubFactory(OrderModelFactory)
+    #product = factory.SubFactory(ProductFactory)
+
+    order_items_qantity = factory.Faker('random_int', min=1, max=100)
```
Comment (v-lavrushko) [inline ProjectShop/order/tests/factories.py:30]: *quantity

File: ProjectShop/order/tests/tests.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        Order = OrderModelFactory()
+        self.assertIsNotNone(Order)
+
+class OrderItemsModelTestCase(TestCase):
+    def setUp(self) -> None:
+        pass
+
+    def test_name(self):
+        OrderItems = OrderItemsModelFactory()
+        self.assertIsNotNone(OrderItems)
```
Comment (v-lavrushko) [inline ProjectShop/order/tests/tests.py:21]: PEP 8 - blank line

File: ProjectShop/Shipping/tests/tests.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+from Shipping.tests.factories import *
+
+class ShippingTestCase(TestCase):
+	def setUp(self) -> None:
+		pass
+
+    def test_create(self):
+        shipping = ShippingFactory()
+        self.assertIsNotNone(shipping)
```
Comment (MykhailoGrynenko) [inline ProjectShop/Shipping/tests/tests.py:12]: It seems like there is an indentation issue. What is the purpose of the setUp method, which does nothing?
Screenshot from 2021-10-04 14-10-50

File: ProjectShop/Shipping/tests/tests.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+        
+
+
+
+
+
```
Comment (MykhailoGrynenko) [inline ProjectShop/Shipping/tests/tests.py:19]: So many blank lines over here.

File: ProjectShop/order/tests/factories.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import factory
+import datetime
+
+from order.models import OrderModel
+from order.models import OrderItemsModel
```
Comment (MykhailoGrynenko) [inline ProjectShop/order/tests/factories.py:5]: from order.models import OrderModel, OrderItemsModel

File: ProjectShop/Shipping/tests/tests.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
+from django.test import TestCase
+from django.db.utils import IntegrityError
+
+from Shipping.tests.factories import *
```
Comment (MykhailoGrynenko) [inline ProjectShop/Shipping/tests/tests.py:4]: I believe importing an asterisk is not the best practice.

**Review by MykhailoGrynenko — CHANGES_REQUESTED** [review]
Changes requested.

File: ProjectShop/Shipping/tests/factories.py [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+from UserApp.tests.factories import UserFactory
+
+
+class ShippingFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = ShippingModel
+
+    user = factory.SubFactory(UserFactory)
+
+    postal_code = factory.Faker('45678')
```
Comment (strangerok1507) [inline ProjectShop/Shipping/tests/factories.py:13]: что за провайдер 45678?

File: ProjectShop/order/tests/factories.py (hunk 7 above)
Comment (CheshirLvova) [reply ProjectShop/order/tests/factories.py]: done)

File: ProjectShop/order/tests/factories.py (hunk 3 above)
Comment (CheshirLvova) [reply ProjectShop/order/tests/factories.py]: fixed

File: ProjectShop/order/tests/tests.py (hunk 1 above)
Comment (CheshirLvova) [reply ProjectShop/order/tests/tests.py]: done)

**Review by v-lavrushko — APPROVED** [review]

File: ProjectShop/Shipping/tests/test_models.py [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        shipping = self.shipping
+        self.assertEqual(shipping.city, shipping.__str__())
+
+    def test_region_str(self):
+        shipping = self.shipping
+        self.assertEqual(shipping.region, shipping.__str__())
+
+    def test_country_str(self):
+        shipping = self.shipping
+        self.assertEqual(shipping.country, shipping.__str__())
```
Comment (Maxgioman) [inline ProjectShop/Shipping/tests/test_models.py:31]: як у нас один і той же метод __str__() видає три різні варіанти в 3 тестах і всі правильні?

**Review by ProkhvatylovArtem — APPROVED** [review]

**Pull request merged**